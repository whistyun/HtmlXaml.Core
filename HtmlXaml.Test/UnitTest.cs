
using ApprovalTests;
using ApprovalTests.Reporters;
using HtmlXaml.Core;
using NUnit.Framework;
using System.Linq;
using System.Threading;
using System.Windows.Documents;

namespace HtmlXaml.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest
    {
        [Test]
        [Apartment(ApartmentState.STA)]
        public void Button()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void CodeBlock()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void InlineCode()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Input()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void List()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void Progres()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void TypicalBlock()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [Apartment(ApartmentState.STA)]
        public void TypicalInline()
        {
            ReplaceManager manager = new();
            var html = Utils.ReadHtml();

            var doc = new FlowDocument();
            doc.Blocks.AddRange(manager.Parse(html));

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
