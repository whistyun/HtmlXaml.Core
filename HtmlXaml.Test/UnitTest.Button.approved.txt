<?xml version="1.0" encoding="utf-16"?>
<FlowDocument xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
  <Paragraph>
    <Button IsEnabled="False">
      <FlowDocumentScrollViewer HorizontalScrollBarVisibility="Disabled" VerticalScrollBarVisibility="Disabled" Padding="0,0,0,0" Margin="0,0,0,0">
        <FlowDocument />
      </FlowDocumentScrollViewer>
    </Button> <Button IsEnabled="False"><FlowDocumentScrollViewer HorizontalScrollBarVisibility="Disabled" VerticalScrollBarVisibility="Disabled" Padding="0,0,0,0" Margin="0,0,0,0"><FlowDocument /></FlowDocumentScrollViewer></Button> <Button IsEnabled="False"><FlowDocumentScrollViewer HorizontalScrollBarVisibility="Disabled" VerticalScrollBarVisibility="Disabled" Padding="0,0,0,0" Margin="0,0,0,0"><FlowDocument><Paragraph>some text</Paragraph></FlowDocument></FlowDocumentScrollViewer></Button> <Button IsEnabled="False"><FlowDocumentScrollViewer HorizontalScrollBarVisibility="Disabled" VerticalScrollBarVisibility="Disabled" Padding="0,0,0,0" Margin="0,0,0,0"><FlowDocument><Section TextAlignment="Center" Tag="Center"><Paragraph Tag="Heading1">title</Paragraph></Section><Paragraph>description1 description2</Paragraph></FlowDocument></FlowDocumentScrollViewer></Button></Paragraph>
</FlowDocument>
